### DISCUSSION TITLE
How to completely disable the splashscreen?

### DISCUSSION THREAD

**Discussion by MulverineX** [ionic-team/capacitor#1341] [Q&A]
I figured out how to do it for electron easily, but for android/ios I can't find configuration anywhere to turn it off, I'm told to call a function to tell capacitor when the app is done loading, but I don't know how to do that within vue (within webpack)

**Comment by solojuve1897** [comment]
I don't know about completely disabling the splashscreen but hiding it when the app is loaded in Vue is simple enough. Add the following line in the mounted hook in the App.vue-file:

SplashScreen.hide()

**Comment by MulverineX** [comment]
Do I need to import something or should it just work?

**Comment by solojuve1897** [comment]
In the same file you add:

import { Plugins } from '@capacitor/core'
const { SplashScreen } = Plugins

**Comment by mlynch** [comment]
Closing as it seems this is resolved.

Let us know if you'd like to completely remove it for some reason (and would love to know why!)

**Comment by nate-eisner** [comment]
@mlynch, I do have a use case in which I would like to completely disable the splash screen. I have a native component that is used initially on launch. (different ViewController in iOS and different Activity in Android) The user is then brought into the Capacitor component from that native piece.

**Comment by davidrock** [comment]
Please, let us know how to REMOVE the splash-screen. It should be an optional not required feature.
Is not logical have to hide something when the app is loaded whether I don`t need this thing at all.

**Comment by jnettome** [comment]
Guys, I've made some progress on it tweaking the SplashScreen.java, overriding those fade params. Has anyone got success on it?

**Comment by KingMatrix1989** [comment]
In React-JS, i have placed *Splashscreen.hide()* before *ReactDom.render* line. Splash duration decreased but still it is visible ~ 1 second. It is may to remove Splash completely?

**Comment by ryandegruyter** [comment]
in BridgeActivity.main you can remove Splash.showOnLaunch(this);
But it's not configurable

**Reply by RajHirani** [reply]
I had tried but it was not worked looking for other solution

**Comment by aparajita** [comment]
You can disable the splash screen by setting `launchDuration: 0` in capacitor.config.json.

**Reply by karm435** [reply]
I think the paramter is `launchShowDuration` and not launchDuration.

Here is the code from Bridge.java

```
public static void showOnLaunch(final BridgeActivity a, CapConfig config) {
    Integer duration = config.getInt(CONFIG_KEY_PREFIX + "launchShowDuration", DEFAULT_LAUNCH_SHOW_DURATION);
    Boolean autohide = config.getBoolean(CONFIG_KEY_PREFIX + "launchAutoHide", DEFAULT_AUTO_HIDE);

    if (duration == 0) {
      return;
    }

    show(a, duration, 0, DEFAULT_FADE_OUT_DURATION, autohide, null, true, config);
  }
```

However even after setting this to 0 in the capacitor.config does not work

```
"plugins": {
    "SplashScreen": {
      "launchShowDuration": 0
    }
  },
```

**Reply by aparajita** [reply]
Right, wrote it without checking.

**Comment by danielehrhardt** [comment]
Something new here?

**Comment by scottwilson312** [comment]
None of these solutions are working for me. This can't possibly be that hard!?

**Reply by karm435** [reply]
You probably don't want to hide the splash screen. Because Apple had made it mandatory last year. https://developer.apple.com/forums/thread/28332

**Comment by jcesarmobile** [comment]
In capacitor 3 you can just don’t install the splash screen plugin.

But it will still show the launch storyboard on iOS and the activity background on android and while the webview load you’ll see a white screen for a brief moment.

**Reply by HStokes** [reply]
I'm on cap 3 and never installed the splash screen plugin.  In fact, the capacitor.config.json file has no "plugins" section at all.
I would also like to eliminate the splash screen since I have my own with a black background, like most mobile games use.

**Comment by LoranRendel** [comment]
Is there a solution now?
